mayastor-test/test_rebuild.js

```diff
@@ -0,0 +1,379 @@
+      ),
+      packageName: 'mayastor_service',
+      serviceName: 'Mayastor',
+      options: {
+        keepCase: true,
+        longs: String,
+        enums: String,
+        defaults: true,
+        oneofs: true,
+      },
+    },
+    common.endpoint
+  );
+}
+
+describe('rebuild tests', function() {
+  var client;
+
+  var ObjectType = {
+    NEXUS: 0,
+    SOURCE_CHILD: 1,
+    DESTINATION_CHILD: 2,
+  };
+
+  function checkState(childType, expectedState) {
+    client.ListNexus({}, (err, res, done) => {
+      if (err) return done(err);
+      assert.lengthOf(res.nexus_list, 1);
+
+      let nexus = res.nexus_list[0];
+      assert.equal(nexus.uuid, UUID);
+
+      if (childType == ObjectType.NEXUS) {
+        assert.equal(nexus.state, expectedState);
+      } else if (childType == ObjectType.SOURCE_CHILD) {
+        assert.equal(nexus.children[0].state, expectedState);
+      } else if (childType == ObjectType.DESTINATION_CHILD) {
+        assert.equal(nexus.children[1].state, expectedState);
+      }
+    });
+  }
+
+  function checkNumRebuilds(expected) {
+    client.ListNexus({}, (err, res, done) => {
+      if (err) return done(err);
+      assert.lengthOf(res.nexus_list, 1);
+
+      let nexus = res.nexus_list[0];
+      assert.equal(nexus.uuid, UUID);
+
+      assert.equal(nexus.rebuilds, expected);
+    });
+  }
+
+  function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  const createNexus = args => {
+    return new Promise((resolve, reject) => {
+      client.createNexus(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const destroyNexus = args => {
+    return new Promise((resolve, reject) => {
+      client.destroyNexus(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const addChild = args => {
+    return new Promise((resolve, reject) => {
+      client.addChildNexus(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const removeChild = args => {
+    return new Promise((resolve, reject) => {
+      client.removeChildNexus(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const startRebuild = args => {
+    return new Promise((resolve, reject) => {
+      client.startRebuild(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const stopRebuild = args => {
+    return new Promise((resolve, reject) => {
+      client.stopRebuild(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const rebuildState = args => {
+    return new Promise((resolve, reject) => {
+      client.getRebuildState(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
```

Comment:
please, have a look at https://github.com/openebs/Mayastor/blob/master/csi/moac/grpc_client.js#L66. I think grpc_promise and its `promisifyAll()` can help you here.

Reply:
Done - modified the test code to use grpc_promise